repo2docker/buildpacks/base.py

```diff
@@ -554,6 +562,19 @@ def render(self):
             for k, v in self.get_build_script_files().items()
         }
 
+        # If get_command returns a list or a tuple, it will be stringified (exec form).
+        # If it returns a string, it will be used as is (shell form).
+        cmd = self.get_command()
+        if isinstance(cmd, tuple):
+            cmd = list(cmd)
+        if isinstance(cmd, list):
+            cmd = str(cmd).replace("'", '"')
+        if not isinstance(cmd, str):
+            raise ValueError(
+                'Method "get_command" of buildpack "%s" must return a string, a list or a tuple.'
+                % self.__class__.__name__
+            )
```

Comment:
- Removed validation as this isn't user facing, just for those making custom buildpacks. My goal is to minimize complexity and ease the long term maintenance and code comprehension by doing that.
- Removed tuples fixup for similar reasons.

```suggestion
        cmd = self.get_command()
        if isinstance(cmd, list):
            cmd = str(cmd).replace("'", '"')
```

I don't understand what is going on with regards to the replacement operation that remains, but I suspect its not reasonable to provide either.